### PR TITLE
add missing loader formats to docs

### PR DESCRIPTION
**Pull request by kecnry**
### Description

This pull request adds missing formats (targeting plugins) to docs:
* footprints
* subsets
* ramp integration
* trace

Note that this also includes a copy of the logic from #4082 that generalizes the logic to build the options in the wireframe.  Whichever goes in second can rebase to exclude that logic from the diff.

Docs build: https://jdaviz--4086.org.readthedocs.build/en/4086/loaders/formats/index.html

### CODE REVIEW

File: docs/loaders/formats/footprint.rst [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+
+- DS9 region files (.reg)
+- FITS region files (.fits)
+- Python objects: `~regions.Region` or `~regions.Regions`
+
+Footprint Label
+===============
+
+Each footprint overlay requires a label for identification in the viewer:
+
```
Comment (MatthewPortman) [inline docs/loaders/formats/footprint.rst]:
```suggestion
When importing a footprint, you can specify a custom label:
```

File: docs/loaders/formats/footprint.rst [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+.. code-block:: python
+
+    jd.load('fov.reg', format='Footprint', footprint_label='Instrument FOV')
+
+If no label is provided, the default label 'default' will be used. The footprint
+label determines which overlay layer the region will be added to in the Footprints
+plugin.
```
Comment (MatthewPortman) [inline docs/loaders/formats/footprint.rst]:
```suggestion
If no label is provided, the default label 'default' will be used. This label specifies the name of the overlay layer that represents the Footprint region which can be modified in the Footprints plugin. 
```
or something like that. I was confused by 'overlay layer the region will be added to' because that implies you can add multiple footprints to a single layer and also that the footprints (and their labels) can be separated from the orientation layer in the UI.

File: docs/conf.py [hunk 3]
```diff
@@ -1168,6 +1168,88 @@ def expand_attr_ref(match):
...
+                            break
+            except Exception:
+                # Skip files that can't be read
+                continue
+
+    # Return extracted list, or fallback to hardcoded list if extraction fails
+    if option_names:
+        return option_names
+    else:
+        return fallbacks.get(option_type, [])
```
Comment (MatthewPortman) [inline docs/conf.py:1264]: Could we validate `extract_loader_options` by checking the results against `fallbacks` and reporting an error/warning if the results don't at least contain those?

Comment (kecnry) [reply docs/conf.py:1264]: we could, but then we have to make sure to keep fallbacks updated, which somewhat defeats the purpose.  The `extract_loader_options` allows the options to be dynamically created based on what pages exist in the docs, so we only need to remember to add an entry in one place and it should get populated.  In fact, we probably could remove fallback entirely or just have it default to an empty list 🤷‍♂️

Comment (MatthewPortman) [reply docs/conf.py:1264]: I was thinking that `fallbacks` (as it is currently defined) is a decent starting point and not something we should worry about updating often. Just that if even those basic options don't get populated then we definitely have a problem. 

If you remove fallbacks though, I think it'd be worth adding a test to check that the output isn't empty. I have a feeling an issue like this would go unnoticed for a long time.

Comment (kecnry) [reply docs/conf.py:1264]: ok, so you want the build to fail if the extracted list doesn't fully contain the fallbacks (to catch the case where either that framework breaks or where a page was deleted that is listed in fallbacks)?  How/where would you propose that logic lives?

Comment (MatthewPortman) [reply docs/conf.py:1264]: Could we add a test to the test suite?

Comment (kecnry) [reply docs/conf.py:1264]: I can try to see if I can trigger the logic outside of the docs build process, but I'm not sure

**Review by cshanahan1 — APPROVED** [review]

File: docs/conf.py (hunk 3 above)
Comment (kecnry) [reply docs/conf.py:1264]: ok, pushed a change that should do this (note that the bulk of that function was copied in another PR already merged that added line lists docs, so now the diff just shows the new block to validate).

File: docs/conf.py [hunk 4]
```diff
@@ -1226,8 +1226,7 @@ def extract_loader_options(option_type):
                                 if title.endswith(' Format'):
                                     title = title[:-7]
                             elif option_type == 'sources':
-                                # Extract name from "Loading from X"
-                                # or "Loading via X" or "Loading X"
+                                # Extract name from "Loading from X" or "Loading via X" or "Loading X"
```
Comment (MatthewPortman) [inline docs/conf.py]: If you rebase, these changes will go away. I applied them in a previous PR because I have a githook for style checking and even though we ignore this file in our check, it triggers in my local one.

Comment (kecnry) [reply docs/conf.py]: ok, I likely chose the wrong thing during conflict handling then.  Still might be easier to just have your hook match what we do here 🤷‍♂️

**Review by MatthewPortman — APPROVED** [review]

**Comment by kecnry** [comment]
failures should be unrelated (and hopefully temporary) - nothing outside of docs is touched here and docs are building successfully
